Sources/SwiftProtobuf/TextFormatEncodingVisitor.swift

```diff
@@ -63,6 +63,23 @@ internal struct TextFormatEncodingVisitor: Visitor {
     self.options = options
   }
 
+  private func formatFieldName(lookingUp fieldNumber: Int) -> [UInt8] {
+      var bytes = [UInt8]()
+      if let protoName = nameMap?.names(for: fieldNumber)?.proto {
+          bytes.append(contentsOf: protoName.utf8Buffer)
+      } else if let protoName = nameResolver[fieldNumber] {
+          let buff = UnsafeBufferPointer(start: protoName.utf8Start, count: protoName.utf8CodeUnitCount)
+          bytes.append(contentsOf: buff)
+      } else if let extensionName = extensions?[fieldNumber]?.protobufExtension.fieldName {
+          bytes.append(UInt8(ascii: "["))
+          bytes.append(contentsOf: extensionName.utf8)
+          bytes.append(UInt8(ascii: "]"))
+      } else {
+          bytes.append(contentsOf: fieldNumber.description.utf8)
+      }
+      return bytes
+  }
```

Comment:
So this is duplicating logic that is split between the next method and the encoder, which seems like it will let bugs sneak in.  Seems like we should figure out how to these two approaches so the logic isn't duplicated.

Reply:
Good point.  I don't see a good way to do this right now, but I'll think about it for a day and see what occurs to me.

Given the paltry performance gains from caching the name lookup, another option would be to just ditch the `formatFieldName` entirely.